subprojects/docs/src/docs/release/notes.md

```diff
@@ -1,3 +1,31 @@
+The Gradle team is pleased to announce Gradle 4.8.
+
+First and foremost, this release of Gradle features [dependency locking](#locking-of-dynamic-dependencies): a mechanism for locking dependency versions which allows builds to become reproducible in the face of dynamic versions or version ranges. 
+Read the [user manual chapter on dependency locking](userguide/dependency_locking.html) to learn how to take advantage of this exciting new feature.
+
+User experience for [incremental annotation processing is improved](#improved-incremental-annotation-processing). 
+Compilation will no longer fail when a processor does something that Gradle detects will not work incrementally. 
+Unused non-incremental processors no longer prevent incremental compilation. 
+Finally, annotation processors are now able to decide dynamically if they are incremental or not.
+ 
+The publishing plugins gets some much-needed improvements in this release:
+
+ * The Signing Plugin now supports [signing all artifacts of a publication](#signing-publications).
```

Comment:
Let's add links to plugin pages in the user guide when mentioning them for the first time.

Reply:
I think such links will distract from folks reading the rest of the notes. I think we should just link to them in the details.